ludwig/models/modules/sequence_decoders.py

```diff
@@ -146,7 +147,7 @@ def decoder_training(self, encoder_output,
         if self.attention_mechanism is not None:
             self.attention_mechanism.setup_memory(
                 encoder_output,
-                memory_sequence_length=sequence_length_3D(encoder_output)
+                memory_sequence_length=self.max_sequence_length
```

Comment:
I may be wrong here, but I believe `memory_sequence_length` should be a vector of integers of size `[batch_size]`, basically the length of each sequence in the batch. The reason is you don't want the attention mechanism to pay attention to the padding vectors.
One easy way here, is you don't want to use `sequence_length_3D` is to calculate the length in the encoder (just counting the nonzero element of the input integers), and adding the calculated length to the output dictionary like:
```
SequenceInputFeature
    def call(self, inputs, training=None, mask=None):
        assert isinstance(inputs, tf.Tensor)
        assert inputs.dtype == tf.int8 or inputs.dtype == tf.int16 or \
               inputs.dtype == tf.int32 or inputs.dtype == tf.int64
        assert len(inputs.shape) == 2

        lengths = tf.math.count_nonzero(inputs, 1)
        inputs_exp = tf.cast(inputs, dtype=tf.int32)
        encoder_output = self.encoder_obj(
            inputs_exp, training=training, mask=mask
        )

        return {'encoder_output': encoder_output, 'lengths': lengths}
```
so that in the decoder you have that value already calculated for you.